sympy/sets/sets.py

```diff
@@ -1738,23 +1738,25 @@ def _intersect(self, other):
         return self.__class__(el for el in self if el in other)
 
     def _complement(self, other):
-        if other is S.Reals:
+        if isinstance(other, Interval):
             nums = sorted(m for m in self.args if m.is_number)
-            syms = [m for m in self.args if m.is_Symbol]
-            # Reals cannot contain elements other than numbers and symbols.
+            if other is S.Reals and nums != []:
```

Comment:
This line will never be executed. The one which is an instance of `Interval` can't be `S.Reals`. If you wan't to check for S.Reals, you should write `other == S.Reals`.